Source/ProjectFirma.Web/Models/AgreementProject.cs

```diff
@@ -0,0 +1,11 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using LtInfo.Common.DesignByContract;
+
+namespace ProjectFirma.Web.Models
+{
+    public partial class AgreementProject : IAuditableEntity
+    {
+        public string AuditDescriptionString => this.Project != null ? this.Project.ProjectName : "NullProject";

```

Comment:
For these types of entities I usually try to include all IDs.
something like: $"AgreementID:{AgreementID}  ProjectID:{ProjectID}"

Reply:
Done. I suppose it's redundant to have AgreementProjectID in there since it's RecordID in the audit? So I can pull that out if you want.
![image](https://github.com/sitkatech/wadnrforesthealth/assets/107082060/10354107-2833-4297-be9f-fbc2a60c83e3)